op-bindings: remove

tynes opened the pull request:
**Description**

Removes the `op-bindings` package from the monorepo. It is now
the responsibility of services to ensure that they are compatible
with particular versions of the contracts. This will reduce the
need for feature branches, as there doesn't have to be a single
implementation of a contract with its bindings.

The following link shows an easy way to generate bindings for
your own service: https://github.com/ethereum-optimism/optimism/blob/943ed0f94f02a2ebad5155334828cc07fca46e0f/op-chain-ops/justfile

`abigen` from go-ethereum is very useful here.

<!-- Contributions welcome! See https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md -->

tynes committed:
op-bindings: remove

Removes the `op-bindings` package from the monorepo. It is now
the responsibility of services to ensure that they are compatible
with particular versions of the contracts. This will reduce the
need for feature branches, as there doesn't have to be a single
implementation of a contract with its bindings.

The following link shows an easy way to generate bindings for
your own service: https://github.com/ethereum-optimism/optimism/blob/943ed0f94f02a2ebad5155334828cc07fca46e0f/op-chain-ops/justfile

`abigen` from go-ethereum is very useful here.

tynes committed:
indexer: remove op-bindings from dockerfile

refcell commented on op-e2e/system_tob_test.go:
```diff
@@ -10,8 +10,8 @@ import (
 
-	legacybindings "github.com/ethereum-optimism/optimism/op-bindings/bindings"
 	"github.com/ethereum-optimism/optimism/op-chain-ops/crossdomain"
+	legacybindings "github.com/ethereum-optimism/optimism/op-e2e/bindings"
```
This should be updated here or in a follow up pr to be `bindings` instead of `legacybindings` now that these are canonical

tynes replied:
Cannot remove all references due to some type issues between packages

refcell commented on op-e2e/withdrawal_helper.go:
```diff
@@ -7,8 +7,8 @@ import (
 
-	legacybindings "github.com/ethereum-optimism/optimism/op-bindings/bindings"
 	"github.com/ethereum-optimism/optimism/op-chain-ops/crossdomain"
+	legacybindings "github.com/ethereum-optimism/optimism/op-e2e/bindings"
```
ibid

tynes committed:
op-e2e: cleanup

Cannot completely remove `legacybindings` because of
dependencies in types in different libraries.
Would be solved by a `go-sdk`.

tynes committed:
Revert "op-e2e: cleanup"

This reverts commit 43791f95213ad9566adeba087a022a48b794dc45.
